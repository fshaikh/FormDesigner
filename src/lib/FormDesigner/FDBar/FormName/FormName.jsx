/**
 * React specific imports
 */
import React from 'react';

/**
 * Import material-ui components
 */
import TextField from '@material-ui/core/TextField';

/**
 * Library specific imports
 */
import FormDesignerContext from '../../Store/FormDesignerContext';
import * as Actions from '../../Actions/Actions';

/**
 * Presentational component for representing Form Name UI
 * Uses TextField component of Material UI for rendering the text field
 * Consumes Context to read form definition and dispatch action when form name value changes
 */
const FormName = () => {
    return (
        <FormDesignerContext.Consumer>
            { (context) => { 
                return <TextField defaultValue={context.formDefinition.formName}
                              placeholder={context.strings.FormNamePlaceholder}
                              onChange={(event) => context.eventEmitter.dispatch(Actions.FormNameChangeAction(event.target.value))}     
                        />  
                }
            }
        </FormDesignerContext.Consumer>
    );
};

export default FormName;