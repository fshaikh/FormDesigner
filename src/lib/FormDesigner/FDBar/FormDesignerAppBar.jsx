/**
 * React specific imports
 */
import React from 'react';

/**
 * MAterial UI specific imports
 */
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
// import MenuIcon from '@material-ui/icons/Menu';

/**
 * Library specific imports
 */
import FormName from './FormName/FormName';

/**
 * Presentation coponent which implements the Top Bar for the Form Designer
 * @param {*} props 
 */
const FormDesignerAppBar = (props) => {
    return (
        <div>
            <AppBar position="static">
                <Toolbar>
                    <Typography variant="title" color="inherit">
                        <FormName />
                    </Typography>
                </Toolbar>
            </AppBar>
        </div>
        
    );
};

export default FormDesignerAppBar;