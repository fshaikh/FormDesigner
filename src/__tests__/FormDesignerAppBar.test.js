import React from 'react';
import renderer from 'react-test-renderer';
import getElementWithContext from 'react-test-context-provider'

import * as Core from './core';
import FormDesignerAppBar from '../lib/FormDesigner/FDBar/FormDesignerAppBar'
import FormName from '../lib/FormDesigner/FDBar/FormName/FormName';


describe('Form Designer App Bar component', () => {
    describe('snapshot tests', () => {
        it('should render correctly', () => {
            const Provider = Core.getContextProvider()
            const component = renderer.create(
                    <Provider value={Core.getContextDefault()}>
                        <FormDesignerAppBar>
                            <FormName />
                        </FormDesignerAppBar>
                    </Provider>
                ).toJSON();
            expect(component).toMatchSnapshot()
        });
    });
});