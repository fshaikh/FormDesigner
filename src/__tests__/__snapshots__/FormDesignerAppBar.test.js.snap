// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form Designer App Bar component snapshot tests should render correctly 1`] = `
<div>
  <header
    className="MuiPaper-root-010 MuiPaper-elevation4-016 MuiAppBar-root-01 MuiAppBar-positionStatic-05 MuiAppBar-colorPrimary-08"
  >
    <div
      className="MuiToolbar-root-037 MuiToolbar-regular-039 MuiToolbar-gutters-038"
    >
      <h2
        className="MuiTypography-root-041 MuiTypography-title-047 MuiTypography-colorInherit-060"
      >
        <div
          aria-describedby={undefined}
          className="MuiFormControl-root-066"
        >
          <div
            className="MuiInput-root-070 MuiInput-formControl-071 MuiInput-underline-074"
          >
            <input
              aria-invalid={false}
              autoComplete={undefined}
              autoFocus={undefined}
              className="MuiInput-input-078"
              defaultValue="New Form"
              disabled={false}
              id={undefined}
              name={undefined}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={undefined}
              onKeyUp={undefined}
              placeholder="Enter form name here"
              readOnly={undefined}
              required={false}
              rows={undefined}
              type="text"
              value={undefined}
            />
          </div>
        </div>
      </h2>
    </div>
  </header>
</div>
`;
